initrd/bin/kexec-boot

```diff
@@ -1,13 +1,89 @@
 #!/bin/sh
 # Launches kexec from saved configuration entries
-set -e -o pipefail
-. /tmp/config
-. /etc/functions
+
+usage() {
+	cat <<End-of-message
+Usage
+=====
+
+Launches kexec from saved configuration entries
+	
+$0 -b /boot -e 'kexec params|...|...' [<options>]
+	
+Options
+-------
+
+### -a <additional confiurations>
+
+Anything provided to the -a argument is appended to the kernel command line (usually comes from parsing the grub, syslinux, BLS files). Ex.
+
+	-a 'export CONFIG_BOOT_KERNEL_ADD="intel_iommu=on intel_iommu=igfx_off"'
+
+### -b <boot path>
+
+path to boot files.  e.g.  -b /boot
+
+### -e <kexec params>
+
+used to prepare parameters passed to kexec system call for booting operating systems on disk.  Fields are pipe separated. 
+
+	"<description>|[elf,xen,multiboot]|kernel <path>|initrd <path>|<parameters>"
+
+First field is ignored.  The second field is the type of booting.  See examples below.
+
+The other fields are kernel, initrd, and parameters to pass to the linux kernel.
+
+EXAMPLE 1: ELF
+
+boot a linux kernel and specify the root partition
+
+	-e "my custom boot|elf|kernel /vmlinuz|initrd /initrd.img|append root=/dev/sda1"
+
+EXAMPLE 2: XEN
+
+boot Qubes, with Xen hypervisor
+
+	-e "xen boot|xen|module /vmlinuz|module /vmlinuz-4.4.67-13.x86_64 placeholder root=/dev/mapper/root ro|module /initramfs-4.4.67-13.x86_64.img"
+
+EXAMPLE 3: MULTIBOOT
+
+boot Qubes, with Xen hypervisor and LUKS encrypted root partition
+
+	-e "Qubes, with Xen hypervisor|multiboot|kernel /xen-4.6.5.gz placeholder |module /vmlinuz-4.4.67-13.pvops.qubes.x86_64 placeholder root=/dev/mapper/luks-UUID ro rd.qubes.hide_all_usb|module /initramfs-4.4.67-13.pvops.qubes.x86_64.img"
+
+Currently, any multiboot entry is interpreted as a Xen-variant and kexec-boot overrides the arguments to the multiboot kernel with custom arguments. 
+
+### -f
+
+some sort of debug mode which prints information about files and sets dryrun on
+
+### -h
+
+show help
+
+### -i
+
+Use this parameter to skip the verification of boot hashes and rollback counter.  This will boot a potentially compromised system.
+
+### -o <initrd>
+
+use the specified initrd for booting
+
+### -r <configurations to remove>
+
+Anything provided to the -r argument is removed from the kernel command line. Ex.
+
+	-r 'export CONFIG_BOOT_KERNEL_REMOVE="quiet"'
```

Comment:
@jtmoree-github-com What I meant to say here (REMOVE and ADD variables, loaded where I pointed to their origin before.) is that they normally come from board configuration.

If you want to explain this, please do not point to export, it makes the user think that this is how they can call those.

in present exemple:
 -r "quiet"
 Would remove "quiet" from the default grub specified options, prior of passing resulting modified boot options to kexec'ed kernel.

Reply:
updated help to use the correct format for -a, -r